src/lib/items/Item.js

```diff
@@ -1,571 +1,615 @@
+      this.props.onSelect(this.itemId, clickType, e)
+    }
+  }
+
+  getItemRef = el => (this.item = el)
+  getDragLeftRef = el => (this.dragLeft = el)
+  getDragRightRef = el => (this.dragRight = el)
+
+
+  getItemProps = (props = {}) => {
+    //TODO: maybe shouldnt include all of these classes
+    const classNames =
+      'rct-item' +
+      (this.props.selected ? ' selected' : '') +
+      (this.canMove(this.props) ? ' can-move' : '') +
+      (this.canResizeLeft(this.props) || this.canResizeRight(this.props)
+        ? ' can-resize'
+        : '') +
+      (this.canResizeLeft(this.props) ? ' can-resize-left' : '') +
+      (this.canResizeRight(this.props) ? ' can-resize-right' : '') +
+      (this.props.item.className ? ` ${this.props.item.className}` : '')
+
+    const dimensions = this.props.dimensions
+
+    
+    const style = {
+      left: `${dimensions.left}px`,
+      top: `${dimensions.top}px`,
+      width: `${dimensions.width}px`,
+      height: `${dimensions.height}px`,
+      lineHeight: `${dimensions.height}px`
+    }
+
+    return {
+      key: this.itemId,
+      ref: this.getItemRef,
+      className: classNames + ` ${props.className? props.className : ''}`,
+      onMouseDown: composeEvents(this.onMouseDown, props.onMouseDown),
+      onMouseUp: composeEvents(this.onMouseUp, props.onMouseUp),
+      onTouchStart: composeEvents(this.onTouchStart, props.onTouchStart),
+      onTouchEnd: composeEvents(this.onTouchEnd, props.onTouchEnd),
+      onDoubleClick: composeEvents(this.handleDoubleClick, props.onDoubleClick),
+      onContextMenu: composeEvents(this.handleContextMenu, props.composeEvents),
+      style: Object.assign({}, props.style, style),
+    }
+  }
+
+  getResizeProps = (props = {}) => {
+    return {
+      left: {
+        ref: this.getDragLeftRef,
+        className: `rct-drag-left ${props.classNameLeft}`
+      },
+      right: {
+        ref: this.getDragRightRef,
+        className: `rct-drag-right ${props.classNameRight}`
+      }
+    }
+  }
+
+  render() {
+    if (typeof this.props.order === 'undefined' || this.props.order === null) {
+      return null
+    }
+
+    const timelineContext = this.context.getTimelineContext()
+    const itemContext = {
+      dimensions: this.props.dimensions,
+      useResizeHandle: this.props.useResizeHandle,
+      title: this.itemDivTitle,
+      canMove: this.canMove(this.props),
+      canResizeLeft: this.canResizeLeft(this.props),
+      canResizeRight: this.canResizeRight(this.props),
+      selected: this.props.selected,
+      dragging: this.state.dragging,
+      dragStart: this.state.dragStart,
+      dragTime: this.state.dragTime,
+      dragGroupDelta: this.state.dragGroupDelta,
+      resizing: this.state.resizing,
+      resizeEdge: this.state.resizeEdge,
+      resizeStart: this.state.resizeStart,
+      resizeTime: this.state.resizeTime,
+      showInnerContentsRender: this.props.dimensions.width > this.props.minimumWidthForItemContentVisibility,

```

Comment:
i think we can change this prop to `width`.  And have the user control when they want to render contents or not.

The `minimumWidthForItemContentVisibility` prop was a stopgap to get some performance out of the timeline.  We can put something in the README for performance such as:

> Performance tip: rendering less DOM is always a good thing.  A tip to improve timelline performance is to only render items whose width is greater than x pixels

Note: you don't need to include this performance section in the README, I can add that.

Reply:
I deleted `minimumWidthForItemContentVisibility` as well